src/main/java/com/iexec/sms/tee/session/base/SecretSessionBaseService.java

```diff
@@ -85,7 +94,9 @@ public SecretSessionBase getSecretsTokens(final TeeSessionRequest request) throw
                     NO_SESSION_REQUEST,
                     "Session request must not be null");
         }
-        if (request.getTaskDescription() == null) {
+        // Task description or deal params should never be null
+        // We nevertheless add both checks to cover NullPointerException
+        if (request.getTaskDescription() == null || request.getTaskDescription().getDealParams() == null) {
             throw new TeeSessionGenerationException(
                     NO_TASK_DESCRIPTION,
                     "Task description must not be null");
```

Comment:
Wouldn't it be better to make the error message more specific? Here, we could have a non-null Task description and still encounter this error.

Reply:
Do you think something like this ?
```suggestion
                    "Task description and deal parameters must not be null");
```

Reply:
Yes 👌